### PR TITLE
*: replace loader by lightning

### PR DESCRIPTION
**Pull request by GregoryIan**
### What is changed, added or deleted? 
we don't recommend to use `loader` any more in TiDB v3.0/3.1/..,  and `lightning` will  take place of `loader`

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [ ] This PR is to align with:
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [ ] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

**Comment by GregoryIan** [comment]
@liukun4515 PTAL

File: dev/TOC.md [hunk 1]
```diff
@@ -60,7 +60,7 @@
   + 运维
     - [Ansible 常见运维操作](/dev/how-to/maintain/ansible-operations.md)
     + 备份与恢复
-      - [使用 Mydumper/Loader 进行备份与恢复](/dev/how-to/maintain/backup-and-restore/mydumper-loader.md)
```
Comment (kissmydb) [inline dev/TOC.md]: 如果 lightning 完全可以代替 loader 的方式，loader 的文档可以去掉，否则建议保留 2 种

File: dev/TOC.md [hunk 2]
```diff
@@ -60,7 +60,7 @@
...
+      - [使用 Mydumper/Loader 进行备份与恢复](/dev/how-to/maintain/backup-and-restore/mydumper-lightning.md)
```
Comment (kissmydb) [inline dev/TOC.md]:
```suggestion
      - [使用 Mydumper/Lightning 进行备份与恢复](/dev/how-to/maintain/backup-and-restore/mydumper-lightning.md)
```

File: dev/TOC.md (hunk 1 above)
Comment (GregoryIan) [reply dev/TOC.md]: 这里是推荐使用，所以只放 lightning

File: v3.0/how-to/maintain/backup-and-restore.md [hunk 3]
```diff
@@ -15,53 +16,24 @@ category: how-to
...
+> PingCAP 研发团队对 `mydumper` 进行了针对 TiDB 的适配性改造，建议使用 PingCAP 官方提供的 [`mydumper`](/v3.0/reference/tools/mydumper.md)。由于使用 `mysqldump` 进行数据备份和恢复都要耗费许多时间，这里也并不推荐。
 
-### `mydumper`/`loader` 全量备份恢复最佳实践
+### `mydumper`/`tidb lightning` 全量备份恢复最佳实践
 
 为了快速地备份恢复数据 (特别是数据量巨大的库)，可以参考以下建议：
 
-* 使用 Mydumper 导出来的数据文件尽可能的小，最好不要超过 64M，可以将参数 `-F` 设置为 64。
-* Loader的 `-t` 参数可以根据 TiKV 的实例个数以及负载进行评估调整，推荐设置为 32。当 TiKV 负载过高，Loader 以及 TiDB 日志中出现大量 `backoffer.maxSleep 15000ms is exceeded` 时，可以适当调小该值；当 TiKV 负载不是太高的时候，可以适当调大该值。
+* 使用 `mydumper` 导出来的数据文件尽可能的小, 可以通过设置参数 `-F` 来控制导出来的文件大小。如果后续使用  TiDB Lightning 对备份文件进行恢复，建议把 `mydumper` -F 参数的值设置为 `256`（单位 MB）；如果使用 `loader` 恢复，则建议设置为 `64`（单位 MB）。
```
Comment (WangXiangUSTC) [inline v3.0/how-to/maintain/backup-and-restore.md]: 这里要不要把 loader 删掉

File: v3.1/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+> **注意：**
+>
+> PingCAP 研发团队对 `mydumper` 进行了针对 TiDB 的适配性改造，建议使用 PingCAP 官方提供的 [`mydumper`](/v3.1/reference/tools/mydumper.md)。由于使用 `mysqldump` 进行数据备份和恢复都要耗费许多时间，这里也并不推荐。
+
+### `mydumper`/`tidb lightning` 全量备份恢复最佳实践
+
+为了快速地备份恢复数据 (特别是数据量巨大的库)，可以参考以下建议：
+
+* 使用 `mydumper` 导出来的数据文件尽可能的小, 可以通过设置参数 `-F` 来控制导出来的文件大小。如果后续使用  TiDB Lightning 对备份文件进行恢复，建议把 `mydumper` -F 参数的值设置为 `256`（单位 MB）；如果使用 `loader` 恢复，则建议设置为 `64`（单位 MB）。
```
Comment (WangXiangUSTC) [inline v3.1/how-to/maintain/backup-and-restore/mydumper-lightning.md]: ditto

File: v3.0/how-to/maintain/backup-and-restore.md [hunk 5]
```diff
@@ -1,11 +1,12 @@
 ---
-title: 备份与恢复
+title: 使用 mydumper/tidb lightning 进行备份与恢复
 category: how-to
+aliases: ['/docs-cn/v3.0/how-to/maintain/backup-and-restore/']
 ---
 
 # 备份与恢复
```
Comment (TomShawn) [inline v3.0/how-to/maintain/backup-and-restore.md]:
```suggestion
# 使用 Mydumper/TiDB Lightning 进行备份与恢复
```

File: v3.0/how-to/maintain/backup-and-restore.md [hunk 6]
```diff
@@ -1,11 +1,12 @@
 ---
-title: 备份与恢复
+title: 使用 mydumper/tidb lightning 进行备份与恢复
```
Comment (TomShawn) [inline v3.0/how-to/maintain/backup-and-restore.md]:
```suggestion
title: 使用 Mydumper/TiDB Lightning 进行备份与恢复
```

File: v3.1/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
+---
+title: 使用 mydumper/tidb lightning 进行备份与恢复
```
Comment (TomShawn) [inline v3.1/how-to/maintain/backup-and-restore/mydumper-lightning.md]:
```suggestion
title: 使用 Mydumper/TiDB Lightning 进行备份与恢复
```

File: v3.1/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+category: how-to
+aliases: ['/docs-cn/v3.1/how-to/maintain/backup-and-restore/mydumper-lightning']
```
Comment (TomShawn) [inline v3.1/how-to/maintain/backup-and-restore/mydumper-lightning.md]:
```suggestion
aliases: ['/docs-cn/v3.1/how-to/maintain/backup-and-restore/mydumper-loader']
```

File: v3.1/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+---
+
+# 备份与恢复
```
Comment (TomShawn) [inline v3.1/how-to/maintain/backup-and-restore/mydumper-lightning.md]:
```suggestion
# 使用 Mydumper/TiDB Lightning 进行备份与恢复
```

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
+---
+title: 使用 mydumper/tidb lightning 进行备份与恢复
```
Comment (TomShawn) [inline dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]:
```suggestion
title: 使用 Mydumper/TiDB Lightning 进行备份与恢复
```

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+category: how-to
+aliases: ['/docs-cn/dev/how-to/maintain/backup-and-restore/mydumper-lightning']
+---
+
+# 备份与恢复
```
Comment (TomShawn) [inline dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]:
```suggestion
# 使用 Mydumper/TiDB Lightning 进行备份与恢复
```

File: dev/TOC.md [hunk 12]
```diff
@@ -60,7 +60,7 @@
   + 运维
     - [Ansible 常见运维操作](/dev/how-to/maintain/ansible-operations.md)
     + 备份与恢复
-      - [使用 Mydumper/Loader 进行备份与恢复](/dev/how-to/maintain/backup-and-restore/mydumper-loader.md)
+      - [使用 Mydumper/TiDB-lightning 进行备份与恢复](/dev/how-to/maintain/backup-and-restore/mydumper-lightning.md)
```
Comment (kennytm) [inline dev/TOC.md]:
```suggestion
      - [使用 Mydumper/TiDB Lightning 进行备份与恢复](/dev/how-to/maintain/backup-and-restore/mydumper-lightning.md)
```

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+本文档将详细介绍如何使用 `mydumper`/`tidb lightning` 对 TiDB 进行全量备份与恢复。增量备份与恢复可使用 [TiDB Binlog](/dev/reference/tidb-binlog/overview.md)。
+
+这里我们假定 TiDB 服务信息如下：
+
+|Name|Address|Port|User|Password|
+|----|-------|----|----|--------|
+|TiDB|127.0.0.1|4000|root|*|
```
Comment (kennytm) [inline dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]: `*` = no password?

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 14]
```diff
@@ -0,0 +1,95 @@
+---
+title: 使用 mydumper/tidb lightning 进行备份与恢复
+category: how-to
+aliases: ['/docs-cn/dev/how-to/maintain/backup-and-restore/mydumper-lightning']
+---
+
+# 备份与恢复
+
+本文档将详细介绍如何使用 `mydumper`/`tidb lightning` 对 TiDB 进行全量备份与恢复。增量备份与恢复可使用 [TiDB Binlog](/dev/reference/tidb-binlog/overview.md)。
```
Comment (kennytm) [inline dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]: Either `tidb-lightning` (the exe name) or "TiDB Lightning" (the product name)

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+
+> **注意：**
+>
+> PingCAP 研发团队对 `mydumper` 进行了针对 TiDB 的适配性改造，建议使用 PingCAP 官方提供的 [`mydumper`](/dev/reference/tools/mydumper.md)。由于使用 `mysqldump` 进行数据备份和恢复都要耗费许多时间，这里也并不推荐。
+
+### `mydumper`/`tidb lightning` 全量备份恢复最佳实践
+
+为了快速地备份恢复数据 (特别是数据量巨大的库)，可以参考以下建议：
+
+* 使用 `mydumper` 导出来的数据文件尽可能的小, 可以通过设置参数 `-F` 来控制导出来的文件大小。如果后续使用  TiDB Lightning 对备份文件进行恢复，建议把 `mydumper` -F 参数的值设置为 `256`（单位 MB）；如果使用 `loader` 恢复，则建议设置为 `64`（单位 MB）。
```
[the remark this answers is not in the document]
Comment (kennytm) [reply dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]: Do we still need to talk about loader?

File: v3.0/how-to/maintain/backup-and-restore.md (hunk 3 above)
Comment (GregoryIan) [reply v3.0/how-to/maintain/backup-and-restore.md]: 这里不要

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md (hunk 13 above)
Comment (GregoryIan) [reply dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]: yep

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md (hunk 14 above)
Comment (GregoryIan) [reply dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]: +1

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+
+这里假定 TiDB 的服务信息如下：
+
+|Name|Address|Port|User|Password|
+|----|-------|----|----|--------|
+|TiDB|127.0.0.1|4000|root|*|
+
+在这个备份恢复过程中，我们会用到下面的工具：
+
+- [Mydumper](/dev/reference/tools/mydumper.md) 从 TiDB 导出数据
```
Comment (TomShawn) [inline dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]:
```suggestion
- [mydumper](/dev/reference/tools/mydumper.md) 从 TiDB 导出数据
```

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+- [TiDB Lightning](/dev/reference/tools/tidb-lightning/overview.md) 导入数据到 TiDB
+
+## 使用 `mydumper`/`TiDB Lightning` 全量备份恢复数据
```
Comment (TomShawn) [inline dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]: TiDB Lightning 大写的话不用加反引号吧？

File: v3.0/how-to/maintain/backup-and-restore.md [hunk 18]
```diff
@@ -15,53 +16,24 @@ category: how-to
 
 在这个备份恢复过程中，我们会用到下面的工具：
 
-- Mydumper 从 TiDB 导出数据
-- Loader 导入数据到 TiDB
+- [Mydumper](/v3.0/reference/tools/mydumper.md) 从 TiDB 导出数据
```
Comment (TomShawn) [inline v3.0/how-to/maintain/backup-and-restore.md]:
```suggestion
- [mydumper](/v3.0/reference/tools/mydumper.md) 从 TiDB 导出数据
```

File: v2.1/TOC.md [hunk 19]
```diff
@@ -260,7 +260,6 @@
       - [常见错误修复](/v2.1/reference/tidb-binlog/troubleshoot/error-handling.md)
     - [FAQ](/v2.1/reference/tidb-binlog/faq.md)
   + 周边工具
-    - [工具使用指南](/v2.1/reference/tools/use-guide.md)
     - [Mydumper](/v2.1/reference/tools/mydumper.md)
```
Comment (TomShawn) [inline v2.1/TOC.md]:
```suggestion
    - [mydumper](/v2.1/reference/tools/mydumper.md)
```

File: v3.1/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+|Name|Address|Port|User|Password|
+|----|-------|----|----|--------|
+|TiDB|127.0.0.1|4000|root|*|
+
+在这个备份恢复过程中，我们会用到下面的工具：
+
+- [Mydumper](/v3.1/reference/tools/mydumper.md) 从 TiDB 导出数据
+- [TiDB Lightning](/v3.1/reference/tools/tidb-lightning/overview.md) 导入数据到 TiDB
+
+## 使用 `mydumper`/`TiDB Lightning` 全量备份恢复数据
```
Comment (TomShawn) [inline v3.1/how-to/maintain/backup-and-restore/mydumper-lightning.md]: 是否要统一一下，mydumper/TiDB Lightning 还是 `mydumper`/`TiDB Lightning`？

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+
+```bash
+./bin/mydumper -h 127.0.0.1 -P 4000 -u root -t 32 -F 256 -B test -T t1,t2 --skip-tz-utc -o ./var/test
+```
+
+上面，我们使用 `-B test` 表明是对 `test` 这个 database 操作，然后用 `-T t1,t2` 表明只导出 `t1`，`t2` 两张表。
+
+`-t 32` 表明使用 32 个线程去导出数据。`-F 256` 是将实际的 table 切分成多大的 chunk，这里就是 256MB 一个 chunk。
+
+`--skip-tz-utc` 添加这个参数忽略掉 TiDB 与导数据的机器之间时区设置不一致的情况，禁止自动转换。
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]: 原句几乎读不通。

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+
+如果 `mydumper` 出现以下报错：
+
+```
+** (mydumper:27528): CRITICAL **: 13:25:09.081: Could not read data from testSchema.testTable: GC life time is shorter than transaction duration, transaction starts at 2019-08-05 21:10:01.451 +0800 CST, GC safe point is 2019-08-05 21:14:53.801 +0800 CST
+```
+
+就再执行两步命令：
+
+1. 执行 `mydumper` 命令前，查询 TiDB 集群的 GC 值并使用 MySQL 客户端将其调整为合适的值。
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]: 读者可能不知道什么是 GC，这里最好能写全 `Garbage Collection (GC)`，或至少需要向用户读者提供可参考的页面。

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 23]
```diff
@@ -0,0 +1,95 @@
...
+
+{{< copyable "shell-regular" >}}
+
+```bash
+./bin/mydumper -h 127.0.0.1 -P 4000 -u root -t 32 -F 256 -B test -T t1,t2 --skip-tz-utc -o ./var/test
+```
+
+上面，我们使用 `-B test` 表明是对 `test` 这个 database 操作，然后用 `-T t1,t2` 表明只导出 `t1`，`t2` 两张表。
+
+`-t 32` 表明使用 32 个线程去导出数据。`-F 256` 是将实际的 table 切分成多大的 chunk，这里就是 256MB 一个 chunk。
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]: `切分成多大的` 不是正式用语。

File: dev/how-to/maintain/backup-and-restore/mydumper-lightning.md [hunk 24]
```diff
@@ -0,0 +1,95 @@
...
+title: 使用 mydumper/tidb lightning 进行备份与恢复
+category: how-to
+aliases: ['/docs-cn/dev/how-to/maintain/backup-and-restore/mydumper-lightning']
+---
+
+# 备份与恢复
+
+本文档将详细介绍如何使用 `mydumper`/`tidb lightning` 对 TiDB 进行全量备份与恢复。增量备份与恢复可使用 [TiDB Binlog](/dev/reference/tidb-binlog/overview.md)。
+
+这里我们假定 TiDB 服务信息如下：
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply dev/how-to/maintain/backup-and-restore/mydumper-lightning.md]: 在技术文档中，应避免使用第一人称“我”、“我们”。

File: v3.0/how-to/maintain/backup-and-restore.md [hunk 25]
```diff
@@ -15,53 +16,24 @@ category: how-to
...
+> PingCAP 研发团队对 `mydumper` 进行了针对 TiDB 的适配性改造，建议使用 PingCAP 官方提供的 [Mydumper](/v3.0/reference/tools/mydumper.md)。由于使用 `mysqldump` 进行数据备份和恢复都要耗费许多时间，这里也并不推荐。
 
-### `mydumper`/`loader` 全量备份恢复最佳实践
+### Mydumper/TiDB Lightning 全量备份恢复最佳实践
 
 为了快速地备份恢复数据 (特别是数据量巨大的库)，可以参考以下建议：
 
-* 使用 Mydumper 导出来的数据文件尽可能的小，最好不要超过 64M，可以将参数 `-F` 设置为 64。
-* Loader的 `-t` 参数可以根据 TiKV 的实例个数以及负载进行评估调整，推荐设置为 32。当 TiKV 负载过高，Loader 以及 TiDB 日志中出现大量 `backoffer.maxSleep 15000ms is exceeded` 时，可以适当调小该值；当 TiKV 负载不是太高的时候，可以适当调大该值。
+* 使用 `mydumper` 导出来的数据文件尽可能的小, 可以通过设置参数 `-F` 来控制导出来的文件大小。如果后续使用  TiDB Lightning 对备份文件进行恢复，建议把 `mydumper` -F 参数的值设置为 `256`（单位 MB）；如果使用 `loader` 恢复，则建议设置为 `64`（单位 MB）。
```
Comment (TomShawn) [inline v3.0/how-to/maintain/backup-and-restore.md]: comment 同 dev

**Pull request merged**